uwblueprint/ccbc
# Closes Forgot Password Flow

## Notion ticket link

[Forgot Password Flow](https://www.notion.so/uwblueprintexecs/Forgot-Password-Flow-41991540c20940c082d4189080294cdc)


NOTE: This PR doesn't add the shapes to the forgot password page. We should make another ticket to add those graphics to all the main pages as needed. 


## Implementation description
* Added a new page called ForgotPassword.tsx
* Modified SetPassword.tsx page to work for both forgot password and create new account
* Modified authService::sendPasswordSetupLink() function to work for both forgot password and new account flow



## Steps to test
1. Ensure you have an account created for our website and that local storage is clear
2. Go to the login page and click forgot password
3. Enter email and click button -> a confirmation modal should be sent
4. Go to your email and follow the steps from the link
5. You should be able to log in with your new password after going through the verification code and set password steps.



## What should reviewers focus on?
* I used the existing create account and password setup infrastructure for the forgot password flow
* I refactored existing functions to make them flexible for both use cases
* Pay attention to the new backend functions I created/ used -> if there is a better way of doing it, I would be happy to discuss alternatives.


## Checklist
